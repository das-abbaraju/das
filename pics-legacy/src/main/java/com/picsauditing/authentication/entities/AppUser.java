package com.picsauditing.authentication.entities;

import org.hibernate.annotations.Cache;
import org.hibernate.annotations.CacheConcurrencyStrategy;

import javax.persistence.*;
import javax.persistence.Entity;
import javax.persistence.Table;
import java.io.Serializable;

@Entity
@Table(name="app_user")
@Cache(usage = CacheConcurrencyStrategy.READ_WRITE, region = "daily")
public class AppUser implements Serializable {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int id;

	private String username;
	private String password = "";
	private String hashSalt;
	private String resetHash;

	public AppUser() {
	}

	public AppUser(int id) {
		this.id = id;
	}

	public AppUser(String username, String password) {
		this.username = username;
		this.password = password;
	}

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getUsername() {
		return username;
	}

	public void setUsername(String username) {
		this.username = username;
	}

	public String getPassword() {
		return password;
	}

	public void setPassword(String password) {
		this.password = password;
	}

	public String getHashSalt() {
		return hashSalt;
	}

	public void setHashSalt(String hashSalt) {
		this.hashSalt = hashSalt;
	}

	public String getResetHash() {
		return resetHash;
	}

	public void setResetHash(String resetHash) {
		this.resetHash = resetHash;
	}
}
