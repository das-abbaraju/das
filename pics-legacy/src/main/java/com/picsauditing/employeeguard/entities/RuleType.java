package com.picsauditing.employeeguard.entities;

public enum RuleType {
	
	REQUIRED("Required", "Required"),
	OPTIONAL("Optional", "Optional");

	private final String displayValue;
	private final String dbValue;

	private RuleType(final String displayValue, final String dbValue) {
		this.displayValue = displayValue;
		this.dbValue = dbValue;
	}

	public String getDisplayValue() {
		return displayValue;
	}

	public String getDbValue() {
		return dbValue;
	}

	public static RuleType fromDbValue(final String dbValue) {
		for (RuleType ruleType : RuleType.values()) {
			if (ruleType.dbValue.equals(dbValue)) {
				return ruleType;
			}
		}

		throw new IllegalArgumentException("Invalid dbValue: " + dbValue);
	}

	@Override
	public String toString() {
		return displayValue;
	}

	public boolean isRequired() {
		return this == REQUIRED;
	}
}
