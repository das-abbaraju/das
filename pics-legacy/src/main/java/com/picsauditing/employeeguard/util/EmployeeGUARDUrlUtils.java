package com.picsauditing.employeeguard.util;

public final class EmployeeGUARDUrlUtils {

	/* Corporate and Operator Links */
	public static final String OPERATOR_SUMMARY = "/employee-guard/operators/dashboard";
	public static final String OPERATOR_ASSIGNMENTS = "/employee-guard/operators/assignments";
	public static final String OPERATOR_PROJECTS = "/employee-guard/operators/projects";
	public static final String OPERATOR_JOB_ROLES = "/employee-guard/operators/role";
	public static final String OPERATOR_SKILLS = "/employee-guard/operators/skill";
	public static final String OPERATOR_PROJECT_CREATE = "/employee-guard/operators/project/create";

	/* Contractor Links */
	public static final String CONTRACTOR_SUMMARY = "/employee-guard/contractor/dashboard";
	public static final String CONTRACTOR_ASSIGNMENTS = "/employee-guard/contractor/project";
	public static final String CONTRACTOR_GROUPS = "/employee-guard/contractor/employee-group";
	public static final String CONTRACTOR_SKILLS = "/employee-guard/contractor/skill";
	public static final String CONTRACTOR_EMPLOYEES = "/employee-guard/contractor/employee";
	public static final String CONTRACTOR_EMPLOYEE_IMPORT = "/contractor/employee/import-export";

	/* Employee Links */
	public static final String IMAGE_LINK = "/employee-guard/employee/contractor/%d/employee-photo/%d";
	public static final String EMPLOYEE_PROFILE = "/employee-guard/employee/profile/%d";
	public static final String EMPLOYEE_SUMMARY = "/employee-guard/employee/dashboard";
	public static final String EMPLOYEE_MY_FILES = "/employee-guard/employee/file";
	public static final String EMPLOYEE_SKILLS = "/employee-guard/employee/skills";

	public static String buildUrl(final String url, final Object... params) {
		return String.format(url, params);
	}

}
