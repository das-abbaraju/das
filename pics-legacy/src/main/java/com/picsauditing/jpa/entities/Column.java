package com.picsauditing.jpa.entities;

import javax.persistence.Entity;
import javax.persistence.Table;
import javax.persistence.Transient;

import com.picsauditing.dr.domain.fields.DisplayType;
import com.picsauditing.report.fields.*;

@Entity
@Table(name = "report_column")
public class Column extends ReportElement implements Comparable<Column> {

	private String columnId;
	private int width = DEFAULT_WIDTH;
	private int sortIndex = DEFAULT_SORT_INDEX;

	public static final int MIN_WIDTH = 35;
	public static final int DEFAULT_WIDTH = 100;
	public static final int DEFAULT_SORT_INDEX = 1;

	public Column() {
	}

	public Column(String fieldName) {
		super(fieldName);
	}

	@Transient
	public String getColumnId() {
		return columnId;
	}

	public void setColumnId(String columnId) {
		this.columnId = columnId;
	}

    @ReportField(type = FieldType.Integer)
	public int getWidth() {
		if (shouldOverrideDefaultWidthWithFieldAnnotation()) {
			width = field.getWidth();
		}

		return width;
	}

	private boolean shouldOverrideDefaultWidthWithFieldAnnotation() {
		return (width == DEFAULT_WIDTH) && (field != null) && (field.getWidth() > 0);
	}

	public void setWidth(int width) {
		this.width = width;
	}

    @ReportField(type = FieldType.Integer)
	public int getSortIndex() {
		return sortIndex;
	}

	public void setSortIndex(int sortIndex) {
		this.sortIndex = sortIndex;
	}

	@Transient
	public DisplayType getDisplayType() {
		if (sqlFunction != null) {
			return sqlFunction.getDisplayType(getField());
		}
		return field.getDisplayType();
	}

	@Override
	public int compareTo(Column otherColumn) {
		return Double.compare(sortIndex, otherColumn.getSortIndex());
	}

}
