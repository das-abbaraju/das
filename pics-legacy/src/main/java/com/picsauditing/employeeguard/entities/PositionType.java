package com.picsauditing.employeeguard.entities;

import com.picsauditing.jpa.entities.Translatable;

public enum PositionType implements Translatable {

	FullTime("FullTime", "Full Time"),
	PartTime("PartTime", "Part Time"),
	Contract("Contract", "Contract");

	private final String displayName;
	private final String dbValue;

	private PositionType(final String dbValue, final String description) {
		this.displayName = description;
		this.dbValue = dbValue;
	}

	public String getDisplayValue() {
		return displayName;
	}

	public String getDbValue() {
		return dbValue;
	}

	public static PositionType fromDbValue(final String dbValue) {
		for (PositionType positionType : PositionType.values()) {
			if (positionType.dbValue.equals(dbValue)) {
				return positionType;
			}
		}

		throw new IllegalArgumentException("Invalid dbValue: " + dbValue);
	}

	@Override
	public String getI18nKey() {
		return this.getClass().getSimpleName() + "." + this.name();
	}

	@Override
	public String getI18nKey(String property) {
		return getI18nKey() + "." + property;
	}

	@Override
	public String toString() {
		return displayName;
	}

}
