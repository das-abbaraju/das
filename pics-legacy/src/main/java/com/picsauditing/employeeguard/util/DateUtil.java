package com.picsauditing.employeeguard.util;

import com.picsauditing.employeeguard.entities.IntervalType;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;

public class DateUtil {
	private static final Logger LOG = LoggerFactory.getLogger(DateUtil.class);

	public static final Date END_OF_TIME = new Date(64060617600000l);

	public static Date explodedToDate(final int year, final int month, final int day) {
		if (year > 0 && month > 0 && day > 0) {
			Calendar calendar = Calendar.getInstance();
			calendar.set(year, month - 1, day);
			return calendar.getTime();
		}

		return null;
	}

	public static int[] dateToExploded(final Date date) {
		if (date != null) {
			int[] exploded = new int[3];
			Calendar calendar = Calendar.getInstance();
			calendar.setTime(date);

			exploded[0] = calendar.get(Calendar.YEAR);
			exploded[1] = calendar.get(Calendar.MONTH) + 1;
			exploded[2] = calendar.get(Calendar.DAY_OF_MONTH);

			return exploded;
		}

		return new int[]{};
	}

	public static boolean doesNotExpire(final Date date) {
		if (date == null) {
			return false;
		}

		if (END_OF_TIME.equals(date)) {
			return true;
		}

		Calendar dayBeforeEndOfTime = Calendar.getInstance();
		dayBeforeEndOfTime.setTime(END_OF_TIME);
		dayBeforeEndOfTime.add(Calendar.DAY_OF_YEAR, -1);

		return date.after(dayBeforeEndOfTime.getTime());
	}

	public static boolean isAboutToExpire(final Date documentExpirationDate, final int intervalPeriod, final IntervalType intervalType) {
		Calendar documentExpiration = Calendar.getInstance();
		documentExpiration.setTime(documentExpirationDate);

		switch (intervalType) {
			case DAY:
				documentExpiration.add(Calendar.DAY_OF_YEAR, -intervalPeriod);
				break;
			case WEEK:
				documentExpiration.add(Calendar.WEEK_OF_YEAR, -intervalPeriod);
				break;
			case MONTH:
				documentExpiration.add(Calendar.MONTH, -intervalPeriod);
				break;
			case YEAR:
				documentExpiration.add(Calendar.YEAR, -intervalPeriod);
				break;
			case NO_EXPIRATION:
				return false;
			case NOT_APPLICABLE:
				return false;
			default:
				throw new IllegalArgumentException();
		}

		return new Date().after(documentExpiration.getTime());
	}

  public static String getSystemTime() {
    SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
    Date systemTime = new Date();

    return dateFormat.format(systemTime);
  }
}
