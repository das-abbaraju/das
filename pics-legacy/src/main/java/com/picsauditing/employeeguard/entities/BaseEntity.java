package com.picsauditing.employeeguard.entities;

import java.io.Serializable;
import java.util.Date;

public interface BaseEntity extends Serializable {
	int getId();
	void setId(int id);

	int getCreatedBy();
	void setCreatedBy(int createdById);

	Date getCreatedDate();
	void setCreatedDate(Date createdDate);

	int getUpdatedBy();
	void setUpdatedBy(int updatedById);

	Date getUpdatedDate();
	void setUpdatedDate(Date updatedDate);

	int getDeletedBy();
	void setDeletedBy(int deletedById);

	Date getDeletedDate();
	void setDeletedDate(Date deletedDate);
}
