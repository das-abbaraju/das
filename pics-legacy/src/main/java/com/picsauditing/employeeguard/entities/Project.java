package com.picsauditing.employeeguard.entities;

import org.hibernate.annotations.Where;

import javax.persistence.*;
import java.util.Date;
import java.util.List;

@Entity
@Table(name = "project")
public class Project implements BaseEntity {
	@Id
	private int id;
	private int accountID;
	private String name;
	private String location;
	private Date startDate;
	private Date endDate;

	@OneToMany(mappedBy = "project", cascade = CascadeType.ALL, orphanRemoval = true)
	@Where(clause = "deletedDate IS NULL AND deletedBy = 0")
	private List<ProjectSkill> skills;

	@OneToMany(mappedBy = "project", cascade = CascadeType.ALL, orphanRemoval = true)
	@Where(clause = "deletedDate IS NULL AND deletedBy = 0")
	private List<ProjectGroup> groups;

	@OneToMany(mappedBy = "project", cascade = CascadeType.ALL, orphanRemoval = true)
	@Where(clause = "deletedDate IS NULL AND deletedBy = 0")
	private List<ProjectCompany> companies;

	private int createdBy;
	private Date createdDate;
	private int updatedBy;
	private Date updatedDate;
	private int deletedBy;
	private Date deletedDate;

	@Override
	public int getId() {
		return id;
	}

	@Override
	public void setId(int id) {
		this.id = id;
	}

	public int getAccountID() {
		return accountID;
	}

	public void setAccountID(int accountID) {
		this.accountID = accountID;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getLocation() {
		return location;
	}

	public void setLocation(String location) {
		this.location = location;
	}

	public Date getStartDate() {
		return startDate;
	}

	public void setStartDate(Date startDate) {
		this.startDate = startDate;
	}

	public Date getEndDate() {
		return endDate;
	}

	public void setEndDate(Date endDate) {
		this.endDate = endDate;
	}

	public List<ProjectSkill> getSkills() {
		return skills;
	}

	public void setSkills(List<ProjectSkill> skills) {
		this.skills = skills;
	}

	public List<ProjectGroup> getGroups() {
		return groups;
	}

	public void setGroups(List<ProjectGroup> groups) {
		this.groups = groups;
	}

	public List<ProjectCompany> getCompanies() {
		return companies;
	}

	public void setCompanies(List<ProjectCompany> companies) {
		this.companies = companies;
	}

	@Override
	public int getCreatedBy() {
		return createdBy;
	}

	@Override
	public void setCreatedBy(int createdBy) {
		this.createdBy = createdBy;
	}

	@Override
	public int getUpdatedBy() {
		return updatedBy;
	}

	@Override
	public void setUpdatedBy(int updatedBy) {
		this.updatedBy = updatedBy;
	}

	@Override
	public int getDeletedBy() {
		return deletedBy;
	}

	@Override
	public void setDeletedBy(int deletedBy) {
		this.deletedBy = deletedBy;
	}

	@Override
	public Date getCreatedDate() {
		return createdDate;
	}

	@Override
	public void setCreatedDate(Date createdDate) {
		this.createdDate = createdDate;
	}

	@Override
	public Date getUpdatedDate() {
		return updatedDate;
	}

	@Override
	public void setUpdatedDate(Date updatedDate) {
		this.updatedDate = updatedDate;
	}

	@Override
	public Date getDeletedDate() {
		return deletedDate;
	}

	@Override
	public void setDeletedDate(Date deletedDate) {
		this.deletedDate = deletedDate;
	}
}
