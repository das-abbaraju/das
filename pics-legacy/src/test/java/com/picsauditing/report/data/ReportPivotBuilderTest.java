package com.picsauditing.report.data;

import com.picsauditing.jpa.entities.Column;
import org.approvaltests.Approvals;
import org.approvaltests.reporters.DiffReporter;
import org.approvaltests.reporters.UseReporter;
import org.junit.Ignore;
import org.junit.Test;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.io.Serializable;
import java.util.List;

@UseReporter(DiffReporter.class)
public class ReportPivotBuilderTest {
    private final Logger logger = LoggerFactory.getLogger(ReportPivotBuilderTest.class);

//    @Test
//    public void testSimplePivot() throws Exception {
//        final List<Column> columns = PivotTestData.getColumnsForFlagChart();
//        final List<List<Serializable>> data = PivotTestData.getDataForFlagChart();
//
//        ReportResults results = ReportResultsFromArrayLists.build(columns, data);
//
//        ReportPivotBuilder builder = new ReportPivotBuilder();
//        ReportResults converted = builder.convertToPivot(results);
//
//        String actual = "Original \n" + results.toString() + "\n\nConverted\n" + converted.toString();
//        Approvals.verify(actual);
//    }
}
