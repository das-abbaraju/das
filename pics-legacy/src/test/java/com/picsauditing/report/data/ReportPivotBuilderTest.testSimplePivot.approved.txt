Original 
ContractorCount	FlagColor	ClientSite	
12	Red	Suncor	
6	Amber	Suncor	
18	Green	Suncor	
4	Red	BASF	
1	Amber	BASF	
12	Green	Tesoro	


Converted
FlagColor	BASF	Suncor	Tesoro	
Amber	1	6	0	
Green	0	18	12	
Red	4	12	0	
